rotary/dataset.py

```diff
@@ -0,0 +1,220 @@
+
+    sample.identifier = sample_identifier
+
+    return sample
+
+
+def generate_dataset_from_sample_tsv(sample_tsv_path, identifier_check=False, integrity_check=False):
+    """
+    Generates a Dataset from a sample TSV file.
+
+    :param sample_tsv_path: Path to the sample TSV file.
+    :param identifier_check: Optional parameter to perform identifier check. Default is False.
+    :param integrity_check: Optional parameter to perform integrity check. Default is False.
+    :return: A Dataset object containing a series of Sample objects.
+
+    :Example:
+    generate_dataset_from_sample_tsv("path/to/sample.tsv", identifier_check=True, integrity_check=True)
+    """
+    dataset = Dataset()
+    with open(sample_tsv_path) as sample_file:
+        tsv_reader = csv.reader(sample_file, delimiter="\t")
+        next(tsv_reader)  # Skip header row.
+        for row in tsv_reader:
+            sample = make_sample_from_sample_tsv_row(row, identifier_check=identifier_check,
+                                                     integrity_check=integrity_check)
+            dataset[sample.identifier] = sample
+
+    return dataset
+
+
+def generate_dataset_from_fastq_directory(input_path):
+    """
+    :param input_path: The path to the directory containing the FASTQ files.
+    :return: A dataset generated from the FASTQ files.
+
+    This method generates a Dataset object from the FASTQ files located in the specified directory. The method first
+    generates a series of SequencingFile objects representing the FASTQ files using the get_fastq_files_in_directory
+    method.
+
+    Then, it organizes the files into groups based on their identifiers using a dictionary called samples_files.
+    Each key in the dictionary represents a unique identifier, and the corresponding value is a list of files
+    associated with that identifier.
+
+    Next, the method iterates through the samples_files dictionary to process each group of sequencing files.
+    It ensures that each group has exactly three files. If not, it raises a ValueError indicating that a sample should
+    have three sequencing files.
+
+    For each group of sequencing files, the method identifies the long_file (a file with no R value), the
+    left_short_file (typically an R1 file), and the right_short_file (typically an R2 file). It creates a
+    Sample object with these files and adds it to the samples list.
+
+    Finally, the method creates a Dataset object using the samples list and returns it.
+    """
+    fastq_files = get_fastq_files_in_directory(input_path)
+
+    samples_files = {}
+    for file in fastq_files:
+        identifier = file.identifier
+        if identifier in samples_files.keys():
+            samples_files[identifier].append(file)
+        else:
+            samples_files[identifier] = [file]
+
+    samples = []
+    for identifier, sequencing_files in samples_files.items():
+        if len(sequencing_files) != 3:
```

Comment:
@jmtsuji Yes, the plan is to eventually have it so that it will be extended to support long read-only configurations. Once the sample object code is moved to its library, it will be extended a bit to support a variety of sample types. For example, my library at our company will be short read-only for 16S sequencing.

Reply:
> am I correct that, in the long term, the generate_dataset_from_fastq_directory function can be extended fairly easily to include long-read only samples?

This will be the eventual plan, but it will be tricky to handle, especially if it's a mix of samples with and without polishing reads. If our code can handle mixed long and hybrid types, we must rewrite the snake make code to process the samples one at a time or dynamically generate the DAG pathing. I can do long read-only right now, but I have to make two fake Illumina files and turn the short read polish flag.

Reply:
> I can do long read-only right now, but I have to make two fake Illumina files and turn the short read polish flag in the config to `False`.

I think we can modify rule `rule set_up_sample_directories:` with some config dependant switching logic and get this functionally

Reply:
Moved to https://github.com/rotary-genomics/rotary/issues/118

Reply:
Cool that this can be extended to a variety of different configurations in the future! Agreed, for rotary, we're going to have to modify the snakefile before we can easily accommodate long read only inputs (see #59 ). For low-hanging fruit, we can 1. make sure that rotary can finish successfully when the short read polish flag is `False` and 2. work on the `set_up_sample_directories` rule like you mentioned. I'll also post another idea in #59 about how to handle the switching logic. I'll resolve this conversation for now... we can continue to discuss in #118 and #59